framework/test/jest/unit/specs/modules/chain/synchronizer/synchronizer.spec.js

```diff
@@ -14,151 +14,411 @@
+				// Arrange
+				const initialLastBlock = {
+					height: genesisBlockDevnet.height + 1,
+					id: 'anId',
+					previousBlockId: genesisBlockDevnet.id,
+					version: 2,
+				};
+				const blocksTempTableEntries = [
+					{
+						...initialLastBlock,
+						fullBlock: initialLastBlock,
+					},
+				];
+				storageMock.entities.TempBlock.get.mockResolvedValue(
+					blocksTempTableEntries,
+				);
+
+				// To load storage tip block into lastBlock in memory variable
+				when(storageMock.entities.Block.get)
+					.calledWith({}, { sort: 'height:desc', limit: 1, extended: true })
+					.mockResolvedValue([initialLastBlock]);
+				await blocksModule.init();
+
+				// Act
+				await synchronizer.init();
+
+				// Assert
+				expect(processorModule.processValidated).not.toHaveBeenCalled();
+				expect(processorModule.deleteLastBlock).not.toHaveBeenCalled();
+				expect(storageMock.entities.TempBlock.truncate).toHaveBeenCalled();
+			});
 		});
-	});
 
-	describe('init()', () => {
-		it('should call restoreBlocksUponStartup if temp_block table is not empty', async () => {
+		it('should not do anything if blocks temporary table is empty', async () => {
 			// Arrange
-			storageMock.entities.TempBlock.isEmpty.mockResolvedValue(false);
+			storageMock.entities.TempBlock.isEmpty.mockResolvedValue(true);
 
 			// Act
 			await synchronizer.init();
 
 			// Assert
-			expect(utils.restoreBlocksUponStartup).toHaveBeenCalled();
+			expect(synchronizerUtils.restoreBlocksUponStartup).not.toHaveBeenCalled();
 		});
 
-		it('should NOT call restoreBlocksUponStartup if temp_block table is empty', async () => {
+		it('should catch any errors and error log it', async () => {
 			// Arrange
-			storageMock.entities.TempBlock.isEmpty.mockResolvedValue(true);
+			const blocksTempTableEntries = new Array(10)
+				.fill(0)
+				.map((_, index) => ({
+					height: index,
+					id: `${index}`,
+					fullBlock: { height: index, id: index, version: 2 },
+				}))
+				.slice(genesisBlockDevnet.height + 2);
+			const initialLastBlock = {
+				height: genesisBlockDevnet.height + 1,
+				id: 'anId',
+				previousBlockId: genesisBlockDevnet.id,
+				version: 1,
+			};
+			storageMock.entities.TempBlock.get.mockResolvedValue(
+				blocksTempTableEntries,
+			);
+			// To load storage tip block into lastBlock in memory variable
+			when(storageMock.entities.Block.get)
+				.calledWith({}, { sort: 'height:desc', limit: 1, extended: true })
+				.mockResolvedValue([initialLastBlock]);
+
+			const error = new Error('error while deleting last block');
+			processorModule.processValidated.mockRejectedValue(error);
+
+			await blocksModule.init();
 
 			// Act
 			await synchronizer.init();
 
 			// Assert
-			expect(utils.restoreBlocksUponStartup).not.toHaveBeenCalled();
+			expect(loggerMock.error).toHaveBeenCalledWith(
+				{ err: error },
+				'Failed to restore blocks from temp table upon startup',
+			);
 		});
 	});
 
 	describe('register()', () => {
```

Comment:
please refactor the synchronizer accordingly and remove these tests since the `register` method is redundant. If you need to move the validations to constructor, however, you can test constructor by providing invalid mechanisms to see if necessary checks are being performed.